ocs_ci/ocs/provider_client_upgrade.py

```diff
@@ -0,0 +1,227 @@
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    skipif_ocp_version,
+    skipif_managed_service,
+    runs_on_provider,
+    skipif_external_mode,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+@skipif_ocs_version("<4.15")
+@skipif_ocp_version("<4.15")
+@skipif_external_mode
+@skipif_managed_service
+@runs_on_provider
+class ProviderUpgrade(OCSUpgrade):
+    """
+    Base class for all provider operator upgrades
+
+    """
+
+    def __init__(
+        self,
+        namespace=constants.OPENSHIFT_OPERATORS,
+        version_before_upgrade=None,
+        ocs_registry_image=None,
+        upgrade_in_current_source=config.UPGRADE.get(
+            "upgrade_in_current_source", False
+        ),
+        resource_name=None,
+    ):
+        if not version_before_upgrade:
+            if config.PREUPGRADE_CONFIG.get("ENV_DATA").get("ocs_version", ""):
+                version_before_upgrade = config.PREUPGRADE_CONFIG["ENV_DATA"].get(
+                    "ocs_version"
+                )
+            else:
+                version_before_upgrade = config.ENV_DATA.get("ocs_version")
+        if not ocs_registry_image:
+            ocs_registry_image = config.UPGRADE.get("upgrade_ocs_registry_image")
+        self.external_cluster = None
+        self.operator_name = None
+        self.subscription_name = None
+        self.pre_upgrade_data = dict()
+        self.post_upgrade_data = dict()
+        self.namespace = namespace
+        # Upgraded phases [pre_upgrade, post_upgrade]
+        self.upgrade_phase = "pre_upgrade"
+        if resource_name:
+            self.resource_name = resource_name
+
+        super().__init__(
+            namespace,
+            version_before_upgrade,
+            ocs_registry_image,
+            upgrade_in_current_source,
+        )
+        self.upgrade_version = self.get_upgrade_version()
+
+    def run_upgrade(self):
+        assert self.get_parsed_versions()[1] >= self.get_parsed_versions()[0], (
+            f"Version you would like to upgrade to: {self.upgrade_version} "
+            f"is not higher or equal to the version you currently running: "
+            f"{self.version_before_upgrade}"
+        )
+
+        self.csv_name_pre_upgrade = self.get_csv_name_pre_upgrade(
+            resource_name=self.resource_name
+        )
+        self.pre_upgrade_images = self.get_pre_upgrade_image(self.csv_name_pre_upgrade)
+        self.load_version_config_file(self.upgrade_version)
+
+        self.channel = self.set_upgrade_channel(resource_name=self.operator_name)
+        self.set_upgrade_images()
+        self.update_subscription(self.channel, self.subscription_name)
+        subscription_plan_approval = config.DEPLOYMENT.get("subscription_plan_approval")
+        if subscription_plan_approval == "Manual":
+            wait_for_install_plan_and_approve(config.ENV_DATA["cluster_namespace"])
+
+        for sample in TimeoutSampler(
+            timeout=725,
+            sleep=5,
```

Comment:
sleep every 5 sec during 10 min will leave too many logs. please increase sleep to ±20-30 sec

Reply:
Addressed.